control-plane/types/src/v0/store/replica.rs

```diff
@@ -97,8 +97,12 @@ impl SpecTransaction<ReplicaOperation> for ReplicaSpec {
     fn commit_op(&mut self) {
         if let Some(op) = self.operation.clone() {
             match op.operation {
-                ReplicaOperation::Unknown => {
-                    panic!("Unknown operation not supported");
+                ReplicaOperation::Unknown => unreachable!(),
```

Comment:
For some of the resources you have used `unimplemented`. We should use the same for all of them.